server/drivers/hd44780-gpiod.c

```diff
@@ -58,29 +80,39 @@ typedef struct {
  * \return          0 on success; -1 on error.
  */
 static int
-init_gpio_pin(Driver *drvthis, struct gpiod_line **pin, const char *name)
+init_gpio_pin(Driver *drvthis, gpio_pin **gpin, const char *name)
 {
 	PrivateData *p = (PrivateData *) drvthis->private_data;
 	gpio_pins *pins = (gpio_pins *) p->connection_data;
-	char config_key[8];
+	char config_key[10];
 	int number;
 
 	snprintf(config_key, sizeof(config_key), "pin_%s", name);
 	number = drvthis->config_get_int(drvthis->name, config_key, 0, -1);
 	if (number == -1)
 		return -1;
 
-	*pin = gpiod_chip_get_line(pins->chip, number);
-	if (*pin == NULL) {
-		report(RPT_ERR, "init_gpio_pin: unable to request GPIO%d: %s",
-		       number, strerror(errno));
-		return -1;
-	}
+	*gpin = calloc(1, sizeof(gpio_pin));
+	gpio_pin *pin = *gpin;
```

Comment:
IIRC in body declarations are not LCDproc coding style.

Anyway, the naming convention between `pin` and `gpin`seems somewhat unintuitive. Probably the easiest to understand solution would be, to just add the asterisk each time to de-reference the double pointer.

If instead you want to change the function to return a pointer directly instead of passing the double pointer, that would also be okay.

(Yes, nitpicking that one. I wouldn't bother, if other changes weren't needed.)